File: sub-music-js/commands/play.js
module.exports = {
  name: "p",
  aliases: ["play"],
  description: "Play a YouTube URL or search query.",

  async execute({ message, args, musicManager }) {
    if (!message.guild) {
      await message.reply("❌ This command can only be used in a server.");
      return;
    }

    const voiceChannel = message.member?.voice?.channel;
    if (!voiceChannel) {
      await message.reply("❌ Join a voice channel first.");
      return;
    }

    const query = args.join(" ").trim();
    if (!query) {
      await message.reply("Usage: `s!p <youtube link or search query>`");
      return;
    }

    const state = musicManager.getGuildState(message.guild.id);

    try {
      await state.connect(voiceChannel);
    } catch (error) {
      console.error("[SUB Music] Voice join failed:", error);
      await message.reply("❌ Could not join your voice channel.");
      return;
    }

    let track;
    try {
      track = await musicManager.resolveTrack(query);
    } catch (error) {
      console.error("[SUB Music] Track resolve failed:", error);
      await message.reply("❌ Failed to fetch this YouTube track.");
      return;
    }

    if (!track) {
      await message.reply("❌ No YouTube results found.");
      return;
    }

    track.requestedBy = message.author.tag;

    const result = await state.enqueueAndMaybePlay(track, message.channel);

    if (result.started) {
      await message.reply(`▶️ Starting playback: **${track.title}**`);
    } else {
      const pending = state.getQueueView().pending.length;
      await message.reply(`✅ Added to queue: **${track.title}** (Position: ${pending})`);
    }
  }
};

File: sub-music-py/commands/play.py
"""Play command for SUB Music."""

from __future__ import annotations

from discord.ext import commands
from utils.ytdl_source import YTDLSource, YTDLSourceError


class Play(commands.Cog):
    """Commands related to starting playback."""

    def __init__(self, bot: commands.Bot) -> None:
        self.bot = bot

    @commands.command(name="p", aliases=["play"])
    async def play(self, ctx: commands.Context, *, query: str | None = None) -> None:
        """Play a YouTube link or search query."""

        if ctx.guild is None:
            await ctx.reply("❌ This command can only be used in a server.")
            return

        if not query:
            await ctx.reply("Usage: `s!p <youtube link or search query>`")
            return

        try:
            state, track, started = await self.bot.music_controller.enqueue(ctx, query)
        except commands.CommandError as exc:
            await ctx.reply(f"❌ {exc}")
            return
        except YTDLSourceError as exc:
            await ctx.reply(f"❌ Could not load that track: {exc}")
            return
        except Exception:
            await ctx.reply("❌ An unexpected error occurred. Please try again.")
            return

        if started:
            await ctx.reply(f"▶️ Starting playback: **{track.title}**")
        else:
            await ctx.reply(f"✅ Added to queue: **{track.title}** (Position: {state.queue.size()})")


async def setup(bot: commands.Bot) -> None:
    await bot.add_cog(Play(bot))
